### PR TITLE
update Azuredisk doc

### PR DESCRIPTION
**Pull request by andyzhangx**
From v1.7.2, we have added support for managed disk in storage class kubernetes.io/azure-disk. This PR is going to update the https://kubernetes.io/docs/concepts/storage/persistent-volumes/#azure-disk doc . The old storage class is still compatible, while we have added a new field "kind" to add support for manage disk.

---
This change is [Reviewable](https://reviewable.kubernetes.io/reviews/kubernetes/kubernetes.github.io/4912)

### CODE REVIEW

**Comment by andyzhangx** [comment]
@karataliu

File: docs/concepts/storage/persistent-volumes.md [hunk 1]
```diff
@@ -695,21 +695,43 @@ parameters:
...
   skuName: Standard_LRS
   location: eastus
   storageAccount: azure_storage_account_name
 ```
 
-* `skuName`: Azure storage account Sku tier. Default is empty.
+* `skuName`: Azure storage account Sku tier. Default is Standard_LRS, allowed values are Standard_LRS and Premium_LRS
 * `location`: Azure storage account location. Default is empty.
-* `storageAccount`: Azure storage account name. If storage account is not provided, all storage accounts associated with the resource group are searched to find one that matches `skuName` and `location`. If storage account is provided, it must reside in the same resource group as the cluster, and `skuName` and `location` are ignored.
+* `storageAccount`: Azure storage account name. If storage account is provided, it must reside in the same resource group as the cluster, and `skuName` and `location` are ignored. If storage account is not provided, a new storage account will be created.
```
Comment (karataliu) [inline docs/concepts/storage/persistent-volumes.md]: `If storage account is not provided, a new storage account will be created.` not accurate. 

Should add '.. in the cluster's resource group if no proper storage matches found.

**Comment by andyzhangx** [comment]
@kyliel @rootfs

File: docs/concepts/storage/persistent-volumes.md (hunk 1 above)
Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: fixed

**Comment by andyzhangx** [comment]
@lachie83 @seanknox

File: docs/concepts/storage/persistent-volumes.md [hunk 2]
```diff
@@ -695,6 +695,8 @@ parameters:
 
 #### Azure Disk
 
+##### Azure Unmanaged Disk Storage Class (before v1.7.2)
```
Comment (rootfs) [inline docs/concepts/storage/persistent-volumes.md]: this is ambiguous :) it is still supported

File: docs/concepts/storage/persistent-volumes.md [hunk 3]
```diff
@@ -711,6 +713,26 @@ parameters:
 * `location`: Azure storage account location. Default is empty.
 * `storageAccount`: Azure storage account name. If storage account is not provided, all storage accounts associated with the resource group are searched to find one that matches `skuName` and `location`. If storage account is provided, it must reside in the same resource group as the cluster, and `skuName` and `location` are ignored.
 
+##### New Azure Disk Storage Class (from v1.7.2)
```
Comment (rootfs) [inline docs/concepts/storage/persistent-volumes.md]: starting from v1.7.2

File: docs/concepts/storage/persistent-volumes.md (hunk 2 above)
Comment (seanknox) [inline docs/concepts/storage/persistent-volumes.md]: Kubernetes users expect a single storage abstraction and ubiquitous access to the storage platform, but Azure Managed Disks dependencies are complex (VMs, premium/non-premium storage, etc). To that end, for Azure Container Service customers we're considering specifying the following scenarios:

- Azure Premium Managed VM can only attach managed disk (both Premium and Standard LRS)
- Azure Managed VM can only attach managed disk (only Standard LRS)
- Unmanaged VM can only attach blob storage disk

For these docs, perhaps it's worth at least mentioning that hybrid clusters with managed disk/blob storage are not recommended. @jdumars, @lachie83, Andy, and I have a task to explore using Azure Instance Metadata Service to apply disk labels to nodes to help with scheduling decisions, but obviously that's not yet an option.

Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: @seanknox I have added the notes in the doc, see:
Premium VM can attach both Standard_LRS and Premium_LRS disks, while Standard VM can only attach Standard_LRS disks.
Managed VM can only attach managed disks and unmanaged VM can only attach unmanaged disks.

Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: @rootfs yes, it's still supported, and compatible after v1.7.2, I have updated the doc

File: docs/concepts/storage/persistent-volumes.md (hunk 3 above)
Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: fixed

**Comment by andyzhangx** [comment]
you could review the preview doc here:
https://deploy-preview-4912--kubernetes-io-master-staging.netlify.com/docs/concepts/storage/persistent-volumes/#azure-file    @jdumars

**Comment by jdumars** [comment]
@andyzhangx LGTM, just needs a rebase

File: docs/concepts/storage/persistent-volumes.md [hunk 4]
```diff
@@ -695,6 +695,8 @@ parameters:
 
 #### Azure Disk
 
+`Azure Unmanaged Disk Storage Class`
```
Comment (lachie83) [inline docs/concepts/storage/persistent-volumes.md]: I would explicitly call out that this is for k8s version <= 1.7.2

File: docs/concepts/storage/persistent-volumes.md [hunk 5]
```diff
@@ -711,6 +713,26 @@ parameters:
 * `location`: Azure storage account location. Default is empty.
 * `storageAccount`: Azure storage account name. If storage account is not provided, all storage accounts associated with the resource group are searched to find one that matches `skuName` and `location`. If storage account is provided, it must reside in the same resource group as the cluster, and `skuName` and `location` are ignored.
 
+`New Azure Disk Storage Class (starting from v1.7.2)`
```
Comment (lachie83) [inline docs/concepts/storage/persistent-volumes.md]: Let's explicitly give examples of all types. Blob, managed standard/premium.

File: docs/concepts/storage/persistent-volumes.md [hunk 6]
```diff
@@ -711,6 +713,26 @@ parameters:
...
+
+```yaml
+kind: StorageClass
+apiVersion: storage.k8s.io/v1
+metadata:
+  name: sharedhdd
```
Comment (lachie83) [inline docs/concepts/storage/persistent-volumes.md]: I think you should keep the name the same as the first example to demonstrate how the format has changed

File: docs/concepts/storage/persistent-volumes.md [hunk 7]
```diff
@@ -711,6 +713,26 @@ parameters:
...
+provisioner: kubernetes.io/azure-disk
+parameters:
+  skuName: Standard_LRS
+  kind: Shared
```
Comment (lachie83) [inline docs/concepts/storage/persistent-volumes.md]: Is skuName still used. I thought it was changed to storageaccounttype

```
  parameters:
    kind: Managed
    storageaccounttype: Standard_LRS
```

File: docs/concepts/storage/persistent-volumes.md (hunk 5 above)
Comment (lachie83) [inline docs/concepts/storage/persistent-volumes.md]: Maybe break out blob from managed and provide a brief overview of why you would use one over the other for users.

File: docs/concepts/storage/persistent-volumes.md (hunk 4 above)
Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: the original yaml file still works, it's compatable

File: docs/concepts/storage/persistent-volumes.md (hunk 6 above)
Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: fixed

File: docs/concepts/storage/persistent-volumes.md (hunk 7 above)
Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: Actually it's the same meaning, user could specify skuName or storageaccounttype, do you think we need to storageaccounttype in the new config?

File: docs/concepts/storage/persistent-volumes.md (hunk 5 above)
Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: Hi @lachie83 I would like to give a link to the github examples when https://github.com/kubernetes/kubernetes/pull/49468 is merged into master branch.

File: docs/concepts/storage/persistent-volumes.md (hunk 7 above)
Comment (lachie83) [inline docs/concepts/storage/persistent-volumes.md]: I think storageaccounttype is has more meaning to a user than skuName. WDYT?

**Comment by zacharysarah** [comment]
@andyzhangx :wave: This looks solid from a docs perspective, but I'd like to make some small edits before merging. Would you please check the box to allow edits from maintainers? Thanks in advance. 🙇  

Allow edits from maintainers checkbox

File: docs/concepts/storage/persistent-volumes.md (hunk 7 above)
Comment (andyzhangx) [inline docs/concepts/storage/persistent-volumes.md]: @lachie83 ok, I have used storageaccounttype in the new storage class.

**Comment by andyzhangx** [comment]
@zacharysarah done, thanks for the check.

**Comment by zacharysarah** [comment]
@andyzhangx Thanks for letting me tinker! Approved with edits. 👍

**Pull request merged**